package.json

```diff
@@ -31,6 +31,7 @@
     "test-ts": "NODE_ENV=test mocha --require ts-node/register --recursive --bail \"src/**/*.test.ts\"",
     "test-watch": "npm run build && mocha --require ts-node/register --watch --recursive \"src/**/*.test.ts\"",
     "prepare": "npm run lint && npm run clean-build && npm test",
+    "preinstall": "npm run compile",
```

Comment:
It is unreasonable to expect that people who want to use `cspell` should need to compile it on install.

Reply:
@Jason3S I agree with you on that point. Still, we need to find a way to make it compile when it needs to be compiled, like when you clone it from source.

I came with some make background where you usually have an install target that depends on build/compile, which would try to build the code if it was not already build. I am not sure how this use case fits the nodejs universe. Clearly this PR does not make sense in its current form but I also don't know what is the right way to address it.